### PR TITLE
DBZ-894 Customize TableId comparison per-connector in filters

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-894

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/Selectors.java [hunk 1]
```diff
@@ -188,17 +189,33 @@ public TableSelectionPredicateBuilder excludeSchemas(String schemaNames) {
          * 
          * @param fullyQualifiedTableNames the comma-separated list of fully-qualified table names to include; may be null or
          *            empty
+         * @param tableIdConverter an arbitrary converter used to convert TableId into String for pattern matching.
+         *         Usually used to remove a component from tableId to simplify patterns.
          * @return this builder so that methods can be chained together; never null
          */
-        public TableSelectionPredicateBuilder includeTables(String fullyQualifiedTableNames) {
+        public TableSelectionPredicateBuilder includeTables(String fullyQualifiedTableNames, Function<TableId, String> tableIdConverter) {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/Selectors.java]: Please let's not use `Function` as parameter type, as it makes it very hard to follow up usages and find references. Instead, please add a dedicated functional interface type. That way, also the purpose and semantics of that type can be documented in a single place.

File: debezium-core/src/main/java/io/debezium/relational/Selectors.java [hunk 2]
```diff
@@ -188,17 +189,33 @@ public TableSelectionPredicateBuilder excludeSchemas(String schemaNames) {
...
             if (isEmpty(fullyQualifiedTableNames)) {
                 tableInclusions = null;
             } else {
-                tableInclusions = Predicates.includes(fullyQualifiedTableNames, TableId::toString);
+                tableInclusions = Predicates.includes(fullyQualifiedTableNames, tableIdConverter);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/Selectors.java]: Looking at its usage, it probably should be rather `tableIdElementConverter` or `tableIdPartConverter`?

**Comment by gunnarmorling** [comment]
Applied; thanks, @jpechane! Note I added one more commit to rename the method of the new interface, so I suppose your PR in _incubator_ will need corresponding adjustment.

**Comment by jpechane** [comment]
@gunnarmorling No need for update on incubator side
